NETDefault/UnitTestingWithNUnit&Moq/MockUserGateWay.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Moq;
+
+namespace UnitTestingExample;
+
+internal sealed class MockUserGateWay : Mock<IUserGateway>
+{
+    private MockUserGateWay(string userName, int? userId = null, string[]? hobbies = null, int? notificationCount = null)
+    {
+        if (userId == null)
+        {
+            Setup(x => x.GetUserIdByUserName(userName))
+                .Throws<NotFoundException>();
+        }
+        else
+        {
+            Setup(x => x.GetUserIdByUserName(userName))
+                .Returns(userId.Value);
+        
+        }
+
+        Setup(x => x.GetHobbiesByUserId(userId ?? -1))
```

Comment:
Does this mean that if we don't add userId we setup a mock for -1? That will lead to some weird behavior right?

Reply:
Well not in this toy example because GetUserIdByUserName is always called, first and already throws an exception when userId is null.